src/App.tsx

```diff
@@ -1,27 +1,95 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import 'bulma/css/bulma.css';
+import classNames from 'classnames';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum Sort {
+  Default = '',
+  All = 'all',
+  FiveFirst = 'fiveFirst',
+  OnlyRed = 'onlyRed',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isChoosedType, setIsChoosedType] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGetGoods = useCallback(async (loadType: string) => {
+    let goodsFromServer: Good[] = [];
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    setIsChoosedType(true);
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    try {
+      switch (loadType) {
+        case Sort.All:
+          goodsFromServer = await getAll();
+          break;
+        case Sort.FiveFirst:
+          goodsFromServer = await get5First();
+          break;
+        case Sort.OnlyRed:
+          goodsFromServer = await getRed();
+          break;
+        default:
+          goodsFromServer = [];
+          break;
+      }
+
+      setGoods(goodsFromServer);
+    } catch {
+      setHasError(true);
+    } finally {
+      setIsChoosedType(false);
+    }
+  }, []);
+
+  return (
+    <div className="App">
+      <h1 className="h1">Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        className={classNames('button',
+          { 'is-loading': isChoosedType })}
+        onClick={() => handleGetGoods('all')}
```

Comment:
```suggestion
        disabled={isLoading}
        onClick={() => handleGetGoods('all')}
```

Disable button to prevent abusing